src/fparser/two/utils.py

```diff
@@ -509,17 +552,12 @@ def match(startcls, subclasses, endcls, reader,
         :rtype: startcls
 
         '''
-        # Have to import C99Preprocessor & Fortran2003 here to avoid circular
-        # import.
-        # pylint: disable=import-outside-toplevel
-        from fparser.two import C99Preprocessor
-        from fparser.two import Fortran2003
         assert isinstance(reader, FortranReaderBase), repr(reader)
```

Comment:
Please add a comment here explaining the use of `di` - i.e. that it is a performance optimisation that saves this method (which is on the hotpath) from having to locally import Fortran2003 and C99Preprocessor.

Reply:
Done